app/main.py

```diff
@@ -1,3 +1,13 @@
 def cache(func):
-    # Write your code here
-    pass
+    container = {}
+
+    def inner_function(*args):
+        if args in container:
+            print("Getting from cache")
+            return container[args]
+        else:
+            result_of_func = func(*args)
+            container[args] = result_of_func
+            print('Calculating new result')
+            return result_of_func
```

Comment:
You can use a single return statement out of the if/else statement instead of two inside it.